src/Build/BuildCheck/Checks/NoEnvironmentVariablePropertyCheck.cs

```diff
@@ -49,6 +49,8 @@ public override void Initialize(ConfigurationContext configurationContext)
 
     public override void RegisterActions(IBuildCheckRegistrationContext registrationContext) => registrationContext.RegisterEnvironmentVariableReadAction(ProcessEnvironmentVariableReadAction);
 
+    internal override bool IsBuiltIn => true;
```

Comment:
for the built-in checks, why the overrides if the base class of `InternalCheck` defaults this member to `true` anyway?

Reply:
Only 'worker node embedded checks' inherit from `InternalCheck`
We might want yet another class denoting `BuiltInCheck` (that the `InternalCheck` would be derived from).

That depends on future decision on if we want to prevent external users from using some Microsoft perceived RuleIds and how we want to implement that (internal overload just for ourselves feel tempting - that would however cut of 1P contributors in their repos)